catalog/nsili/catalog-nsili-source/src/main/java/org/codice/alliance/nsili/source/NsiliSource.java

```diff
@@ -261,25 +262,44 @@
      * Constructor used for testing.
      */
     NsiliSource(SecureCxfClientFactory factory, HashMap<String, String[]> resultAttributes,
-            HashMap<String, List<String>> sortableAttributes, NsiliFilterDelegate filterDelegate) {
+            HashMap<String, List<String>> sortableAttributes, NsiliFilterDelegate filterDelegate,
+            ORB orb) {
         scheduler = Executors.newSingleThreadScheduledExecutor();
         this.factory = factory;
         this.resultAttributes = resultAttributes;
         this.nsiliFilterDelegate = filterDelegate;
         this.sortableAttributes = sortableAttributes;
         ddfOrgName = System.getProperty("org.codice.ddf.system.organization", DEFAULT_USER_INFO);
-        initOrb();
+        this.orb = orb;
     }
 
-    public NsiliSource() {
+    public NsiliSource(CorbaOrb corbaORB) {
         scheduler = Executors.newSingleThreadScheduledExecutor();
+        setCorbaOrb(corbaORB);
     }
 
     public void init() {
+        corbaOrb.addCorbaServiceListener(this);
         initCorbaClient();
         setupAvailabilityPoll();
     }
 
+    @Override
+    public void corbaInitialized() {
+        orb = corbaOrb.getOrb();
+        initCorbaClient();
+    }
+
+    @Override
+    public void corbaShutdown() {
+        shutdownCorbaClient();
+    }
+
+    public void shutdownCorbaClient() {
```

Comment:
does this function really need to exist?  or can the logic just get moved to corbaShutdown? seems redundant